tests/core/test_validation_config.py

```diff
@@ -160,6 +195,65 @@ def test_returns_expected_data(
             },
         )
 
+    @mock.patch.object(ValidationsStore, "set")
+    @pytest.mark.unit
+    def test_persists_validation_results_for_non_cloud(
+        self,
+        mock_validations_store_set: MagicMock,
+        mock_validator: MagicMock,
+        validation_config: ValidationConfig,
+    ):
+        validation_config.suite.add_expectation(
+            gxe.ExpectColumnMaxToBeBetween(column="foo", max_value=1)
+        )
+        mock_validator.graph_validate.return_value = [
+            ExpectationValidationResult(success=True)
+        ]
+
+        validation_config.run()
+
+        mock_validator.graph_validate.assert_called_with(
+            configurations=[
+                ExpectationConfiguration(
+                    expectation_type="expect_column_max_to_be_between",
+                    kwargs={"column": "foo", "max_value": 1.0},
+                )
+            ],
+            runtime_configuration={"result_format": "SUMMARY"},
+        )
+
+        # validate we are calling set on the store with data that's roughly the right shape
+        [(_, kwargs)] = mock_validations_store_set.call_args_list
+        key = kwargs["key"]
+        value = kwargs["value"]
+        assert isinstance(key, ValidationResultIdentifier)
+        assert key.batch_identifier == BATCH_ID
+        assert value.success is True
+
+    @mock.patch.object(ValidationsStore, "set")
+    @pytest.mark.unit
+    def test_persists_validation_results_for_cloud(
+        self,
+        mock_validations_store_set: MagicMock,
+        mock_validator: MagicMock,
+        cloud_validation_config: ValidationConfig,
+    ):
+        cloud_validation_config.suite.add_expectation(
+            gxe.ExpectColumnMaxToBeBetween(column="foo", max_value=1)
+        )
+        mock_validator.graph_validate.return_value = [
+            ExpectationValidationResult(success=True)
+        ]
+
+        cloud_validation_config.run()
+
+        # validate we are calling set on the store with data that's roughly the right shape
+        [(_, kwargs)] = mock_validations_store_set.call_args_list
```

Comment:
ooh I've never seen this syntax before - I guess just an alternative way of unpacking huh (given the shape of the call args list)

Reply:
yup. I kinda prefer this because it implicitly asserts that there was only one call made. Otherwise, I see the use of `call_args_list` getting really verbose with length checking, and then pulling out args/kwargs on their own lines.